pinot-segment-local/src/main/java/org/apache/pinot/segment/local/indexsegment/immutable/ImmutableSegmentLoader.java

```diff
@@ -169,7 +158,20 @@ public static ImmutableSegment load(File indexDir, IndexLoadingConfig indexLoadi
 
     ImmutableSegmentImpl segment =
         new ImmutableSegmentImpl(actualSegmentDirectory, segmentMetadata, indexContainerMap, starTreeIndexContainer);
-    LOGGER.info("Successfully loaded segment {} with config: {}", segmentName, segmentDirectoryLoaderConfigs);
+    LOGGER.info("Successfully loaded segment {} with config: {}", segmentName, segDirConfigs);
     return segment;
   }
+
+  // Pre-process the segment on local using local SegmentDirectory.
+  // Please note that this step may modify the segment metadata.
+  private static void preprocessSegment(File indexDir, IndexLoadingConfig indexLoadingConfig, Schema schema)
+      throws Exception {
+    PinotConfiguration tierConfigs = indexLoadingConfig.getTierConfigs();
+    PinotConfiguration segDirConfigs = new PinotConfiguration(tierConfigs.toMap());
+    SegmentDirectory segDir =
+        SegmentDirectoryLoaderRegistry.getLocalSegmentDirectoryLoader().load(indexDir.toURI(), segDirConfigs);
```

Comment:
Do we need tier configs when loading from local?

Reply:
looks LocalSegmentDirectory just needs ReadMode to init, and getTierConfigs() has check on this property.
but cc @npawar to chime in, in case I missed anything

Reply:
you're right in terms of what the LocalSegmentDirectory needs. I choose to make the `getTierConfigs` be the call that returns all tier related properties. `loadMode` is a tier config for `local` tier. This way I could keep the `getLocalSegmentDirectory` call's arguments consistent across the tiers.

Reply:
@npawar Any reason we need to do `new PinotConfiguration(tierConfigs.toMap())`? Wouldn't it generate the same config?

Reply:
I assume this is to create a new independent copy, probably to avoid accidental writes to the config.

Reply:
I think i wanted to make it clear that segmentDirectoryConfigs are made using tierConfigs, and they are not directly equivalent. But they are same for practical purposes of this PR. Feel free to change it if you wish